### ISSUE TITLE
[Vue 3] `useForm` composable unexpectedly overwrites defaults after a successful form request

### ISSUE THREAD

**Issue by shengslogar** [inertiajs/inertia#1741]
### Version:

- `@inertiajs/vue3` version: 1.0.14

### Describe the problem:

`useForm({}).reset()` is not functional after a successful submission. This is due to the form's internal `onSuccess` hook setting defaults to the current form state (line 172).

While this may make sense when updating existing resources (e.g. after a `PATCH /users/2` request, it would be nice to have defaults updated to any newly-saved state), this doesn't make sense for a form that creates new resources (e.g. `POST /users`) — i.e. there's no way to get the form back to its original state without calling `form.defaults({...}) && form.reset()` each time.

https://github.com/inertiajs/inertia/blob/0fc4b1c8b91639ed58b44fafe8833f2108496de6/packages/vue3/src/useForm.ts#L163-L175

I was initially debating opening a discussion question, but peeking at the source code for [React](https://github.com/inertiajs/inertia/blob/0fc4b1c8b91639ed58b44fafe8833f2108496de6/packages/react/src/useForm.ts#L105-L123) and [Svelte](https://github.com/inertiajs/inertia/blob/0fc4b1c8b91639ed58b44fafe8833f2108496de6/packages/svelte/src/useForm.js#L124-L135), as well as looking at the expected behavior of a [previous discussion question](https://github.com/inertiajs/inertia/discussions/429#discussioncomment-280547) leads me to believe this is indeed a bug. [Older versions of the Vue codebase](https://github.com/inertiajs/inertia/blob/f10e86d04753f859fde85fbb1dfdff06e06775c7/packages/inertia-vue3/src/useForm.js#L100-L111) also do not have this line of code.

### Steps to reproduce:

**1. Set up a new form**

```
import { useForm } from '@inertiajs/vue3'

const form = useForm({ email: '' })
```

**2. Dirty-ify form**

```
form.email = 'foo@bar.com`
```

**3. Submit form**
```
form.post/patch/etc...
```

At this point, assuming the form request was successful, `form.isDirty === false && form.email === 'foo@bar.com`. `form.reset()` yields no changes.

### Workaround:

```
import { useForm } from '@inertiajs/vue3'

const formDefaults = { email: '' }
const form = useForm(formDefaults)

function resetForm() {
    form.defaults(formDefaults)
    form.reset()
}
```

**Comment by WillRy** [comment]
Hello everyone,

I came across the same problem today, it really seems like a bug

**Comment by khaines0625** [comment]
This is a bug.  A fix would be appreciated!

**Comment by pedroborges** [comment]
This has been fixed in #1935.

**Comment by erip2** [comment]
The fix for this is reverted as it says on the last update of the PR.

I think this issue should be opened again and discussed by the team.

If the behavior is different on React, Svelve, and Vue, then a change should be made to make it consistent.

This is causing also an issue for me and was unexpected and something not explained in the documentation.
